include/Dialect/Poly/IR/PolyOps.td

```diff
@@ -65,13 +65,13 @@ def Poly_GetCoeffOp : Poly_Op<"get_coeff"> {
   let summary = "Gets the coefficient of X^{index} from a polynomial.";
 
   let arguments = (ins
-    Polynomial:$x,
-    Index:$index
+    PolynomialLike:$x,
+    IndexLike:$index
```

Comment:
My feeling here is that this might not need `*Like` operands. I can think of cases where we would want to extract a coefficient of a single polynomial, or perhaps the `i`th coefficient of a list, but I can't think of one where we'd want to extract element-wise. (I'm not sure if this was intended to have the trait?)

The cases I can think of `get_coeff` being useful are things like "sample extraction", or conversion from RLWE to LWE ciphertext. In this case if you had some RLWE ciphertext (let's say of dimension 2) of the form `(a, b)` with `a = a_0 + a_1 * x + ... + a_{n-1} * x^{n-1}` and  `b = b_0 + b_1 * x + .. + b_{n-1} * x^{n-1}`, then your LWE ciphertext is `(a_0, -a_{n-1}, ... , -a_1, b_0)`. 

The first coefficients from `a` could probably be made by extracting a slice, negating, and reversing, then appending the first coefficient. Then you'd insert the the extracted 0th coefficient of `b`. So, I could see a `0`th coefficient extraction of the Tensor of `(a, b)`, but I'm not certain that's a big gain over individual extractions of them.

WDYT for simplifying until we have a need for the `*Like` types? @AlexanderViand-Intel let me know if something else comes to mind for extraction